### PR TITLE
Report all stake-pools in GetStakeSnapshots query.

### PR DESCRIPTION
**Pull request by newhoggy**
# Description

Previously, the code only selected pools that hard `Mark` stake.  It needs to report pools that have any of `Mark`, `Go`

Resolves https://github.com/input-output-hk/cardano-node/issues/4895

# Checklist

- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [ ] The documentation has been properly updated
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] Any changes affecting Consensus packages must have an entry in the appropriate `changelog.d` directory created using [`scriv`](https://github.com/input-output-hk/scriv). If in doubt, see the [Consensus release process](../ouroboros-consensus/docs/ReleaseProcess.md).
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [ ] Reviewer requested

### CODE REVIEW

**Review by Jasagredo — APPROVED** [review]
This looks good and given it is a requirement, I see no objections to it

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs [hunk 1]
```diff
@@ -322,8 +322,14 @@ instance (ShelleyCompatible proto era, ProtoCrypto proto ~ crypto) => QueryLedge
...
             Nothing ->
+              let poolIds = Set.fromList $ mconcat
+                    [ VMap.elems (SL._delegations _pstakeMark)
+                    , VMap.elems (SL._delegations _pstakeSet)
+                    , VMap.elems (SL._delegations _pstakeGo)
+                    ]
+              in
               StakeSnapshots
-                { ssStakeSnapshots = getPoolStakes (Set.fromList (VMap.elems (SL._delegations _pstakeMark)))
+                { ssStakeSnapshots = getPoolStakes poolIds
```
Comment (dnadales) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:332]: Hmm, we changed the returned values in `ssStakeSnapshots`. Shouldn't we update the haddocks for this field?

Comment (newhoggy) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:332]: What comes back in that field depends on what is queried for.

**Pull request merged**